src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1523,10 +1523,39 @@ private IEnumerable<ITypeSymbol> InferTypeInYieldStatement(YieldStatementSyntax
 
             private IEnumerable<ITypeSymbol> InferTypeForReturnStatement(ReturnStatementSyntax returnStatement, SyntaxToken? previousToken = null)
             {
+                bool isAsync;
+                IEnumerable<ITypeSymbol> types;
+
+                InferTypeForReturnStatement(returnStatement, previousToken, out isAsync, out types);
+
+                if (!isAsync)
+                {
+                    return types;
+                }
+
+                var taskOfT = this.Compilation.TaskOfTType();
+                if (taskOfT == null || types == null)
+                {
+                    return SpecializedCollections.EmptyEnumerable<ITypeSymbol>();
```

Comment:
"If `types` is `null`, return an empty `IEnumerable`" seems inconsistent, because if `types` _can_ be null, that means on line 1533 above this method could potentially return null, but here that's explicitly turned into an empty collection.